### ISSUE TITLE
False reporting of dictionary comprehension

### ISSUE THREAD

**Issue by yomichi** [tonyhffong/Lint.jl#14]
A test code is as follows.

```
shell> cat lint_test.jl
[i for i in 1:2]
[j => j*j for j in 1:2 ]

julia> lintfile("lint_test.jl")
       lint_test.jl [                    ]    1 ERROR  Use of undeclared symbol j
```
